### PR TITLE
kdenlive: set run-time dep paths (fix #83885)

### LINKED ISSUE #83885: kdenlive doesn't find effects
**Describe the bug**

kdenlive doesn't find most of the effects. On console it's complaining about frei0r with numerous lines similar to:

```
Unknown asset :  "frei0r.sigmoidaltransfer"
```

For instance, Motion->Speed effect is missing.

**To Reproduce**

1. In `configuration.nix`, add:
```
  config.extraPackages = with pkgs; [
    kdenlive
    ffmpeg-full
    frei0r
  ];
```
2. Run `sudo nixos-rebuild test`.
3. Run `kdenlive`.

**Expected behavior**

I expect the normal set of effects to be available.

**Metadata**

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.28, NixOS, 20.09.git.3320a06049f (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - nixpkgs: `/etc/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

There's no maintainer marked in nixpkgs, but @lambda-11235 did some work on #66220.

### PR DESCRIPTION
**Pull request by jluttine**
###### Motivation for this change

See #83885 and #29614. Without these fixes, at least for me, kdenlive was complaining on the command line that it doesn't find ffmpeg nor frei0r. Not sure if it had any real effect on the program otherwise though, but at least those suspicious warnings.

Fixes #83885. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jluttine** [comment]
I don't understand the check failure: `Failed during the setup of executing nix-env: Os { code: 2, kind: NotFound, message: "No such file or directory" }`.

And regarding the missing speed effect: I found speed adjustments by right clicking a clip on the timeline.

File: pkgs/applications/kde/kdenlive.nix [hunk 1]
```diff
@@ -70,14 +70,25 @@ mkDerivation {
...
     ''
       sed -i CMakeLists.txt -e '/find_package(Qt5 REQUIRED/ s|)| Concurrent)|'
       substituteAllInPlace src/kdenlivesettings.kcfg
     '';
+  # Frei0r path needs to be set too or Kdenlive will complain. See:
+  # https://github.com/NixOS/nixpkgs/issues/83885
+  # https://github.com/NixOS/nixpkgs/issues/29614#issuecomment-488849325
+  postInstall = ''
+    wrapProgram $out/bin/kdenlive --set FREI0R_PATH ${frei0r}/bin/frei0r-1
+    wrapProgram $out/bin/kdenlive_render --set FREI0R_PATH ${frei0r}/bin/frei0r-1
```
Comment (Mic92) [inline pkgs/applications/kde/kdenlive.nix:90]: Do avoid double wrapping I would use `qtWrapperArgs+=(--set FREI0R_PATH ${frei0r}/bin/frei0r-1)` instead.

Comment (jluttine) [reply pkgs/applications/kde/kdenlive.nix]: Thanks for the idea! I did it a bit differently now but with the same goal.

**Comment by tobiasBora** [comment]
Thanks for this PR, but could this be backported to the stable branch? This error is still here on 20.03, but if I install kdenlive from unstable it works greatly.

**Pull request merged**